lib/Textile/API/index.ts

```diff
@@ -60,8 +69,19 @@ class API {
     return JSON.parse(result) as File
   }
 
-  addThread = async (key: string, name: string, shared: boolean): Promise<ThreadInfo> => {
-    const result = await TextileNode.addThread(key, name, shared)
+  addThread = async (key: string, name: string, type: ThreadType, sharing: ThreadSharing, members: string[], schema_type: SchemaType, json_schema?: string): Promise<ThreadInfo> => {
+    const stringMembers = members.join(',')
+    let media = false
```

Comment:
How about `const media = schema_type === SchemaType.MEDIA` etc? Less lines, easier to understand.

Reply:
right, will do. that was the result of an evolution of types etc, never got back to making nice.